src/pugixml.cpp

```diff
@@ -6077,6 +6094,22 @@ namespace pugi
 		return true;
 	}
 
+	PUGI__FN bool xml_node::remove_children()
+	{
+		for (xml_node_struct* child = this->_root->first_child; child; child = child->next_sibling)
+		{
+			if (!_root || !child || child->parent != _root) return false;
+
+			impl::xml_allocator& alloc = impl::get_allocator(_root);
```

Comment:
This will return the same object for all nodes, it should be hoisted outside of the loop (along with reserve)